File: src/main/java/br/com/leonardoferreira/jirareport/service/LeadTimeConfigService.java
package br.com.leonardoferreira.jirareport.service;

import br.com.leonardoferreira.jirareport.domain.LeadTimeConfig;
import br.com.leonardoferreira.jirareport.domain.request.LeadTimeConfigRequest;
import br.com.leonardoferreira.jirareport.domain.response.LeadTimeConfigResponse;
import java.util.List;

public interface LeadTimeConfigService {

    List<LeadTimeConfig> findAllByBoardId(Long boardId);

    List<LeadTimeConfigResponse> findAll(Long boardId);

    Long create(Long boardId, LeadTimeConfigRequest leadTimeConfigRequest);

    LeadTimeConfigResponse findByBoardAndId(Long boardId, Long id);

    void update(Long boardId, Long id, LeadTimeConfigRequest leadTimeConfigRequest);

    void deleteByBoardAndId(Long boardId, Long id);
}

File: src/main/kotlin/br/com/jiratorio/service/leadtime/LeadTimeConfigService.kt
package br.com.jiratorio.service.leadtime

import br.com.jiratorio.domain.request.LeadTimeConfigRequest
import br.com.jiratorio.domain.response.LeadTimeConfigResponse

interface LeadTimeConfigService {

    fun findAll(boardId: Long): List<LeadTimeConfigResponse>

    fun create(boardId: Long, leadTimeConfigRequest: LeadTimeConfigRequest): Long

    fun findByBoardAndId(id: Long, boardId: Long): LeadTimeConfigResponse

    fun update(id: Long, boardId: Long, leadTimeConfigRequest: LeadTimeConfigRequest)

    fun deleteByBoardAndId(boardId: Long, id: Long)

}
